test/core/utils/filter-html-attrs.js

```diff
@@ -0,0 +1,246 @@
+describe('axe.utils.filterHtmlAttrs', function() {
+  'use strict';
+  var filterHtmlAttrs = axe.utils.filterHtmlAttrs;
+  var html = '<label> My Label <input type="text" value="my value" /></label>';
+  var expected = '<label> My Label <input value="my value" /></label>';
+
+  it('should return string if no attributes are passed', function() {
+    assert.equal(filterHtmlAttrs(html), html);
+  });
+
+  it('should filter attribute if exists', function() {
+    assert.equal(filterHtmlAttrs(html, { type: true }), expected);
+  });
+
+  it('should filter attribute if matches value', function() {
+    assert.equal(filterHtmlAttrs(html, { type: 'text' }), expected);
+  });
+
+  it('should not filter attribute if does not match value', function() {
+    assert.equal(filterHtmlAttrs(html, { type: 'submit' }), html);
+  });
+
+  it('should filter attribute if matches list of values', function() {
+    assert.equal(filterHtmlAttrs(html, { type: ['submit', 'text'] }), expected);
+  });
+
+  it('should not filter attribute if does not match list of values', function() {
+    assert.equal(filterHtmlAttrs(html, { type: ['submit', 'password'] }), html);
+  });
+
+  it('should filter attribute if node matches nodeName matcher', function() {
+    assert.equal(
+      filterHtmlAttrs(html, {
+        type: {
+          nodeName: 'input'
```

Comment:
Even though I know it isn't, this looks like a bug. I think we can fix that by changing the name.

```suggestion
          forElement: 'input'
```